### PR TITLE
fix: rewrite stripe + bug in unsubscribing

### PR DESCRIPTION
**Pull request by ShubhamPalriwala**
## What does this PR do?
- Fixes bug where the onConfirm & onDiscard got replaced in the PricingCard Alert
- Rewrites Stripe webhook listener to support more complicated cases of unsubscribing

## How should this be tested?
Run the stripe cli locally, authenticate, switch to test mode and start testing various cases!

### Required

- [ ] Filled out the "How to test" section in this PR
- [ ] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: apps/web/app/(app)/environments/[environmentId]/settings/billing/actions.ts [hunk 1]
```diff
@@ -24,6 +25,8 @@ export async function upgradePlanAction(
   if (!isAuthorized) throw new AuthorizationError("Not authorized");
 
   const subscriptionSession = await createSubscription(teamId, environmentId, priceLookupKeys);
+
+  teamCache.revalidate({ id: teamId });
```
Comment (mattinannt) [inline apps/web/app/(app)/environments/[environmentId]/settings/billing/actions.ts:29]: why do we need to revalidate the cache here? We should only revalidate the cache after making the database call that changes the team object. Where is the call made? Can we move the `revalidate` command after the db call instead?

File: apps/web/app/(app)/environments/[environmentId]/settings/billing/actions.ts [hunk 2]
```diff
@@ -58,5 +61,6 @@ export async function removeSubscriptionAction(
 
   const removedSubscription = await removeSubscription(teamId, environmentId, priceLookupKeys);
 
+  teamCache.revalidate({ id: teamId });
```
Comment (mattinannt) [inline apps/web/app/(app)/environments/[environmentId]/settings/billing/actions.ts:64]: same concern as above

**Review by mattinannt — COMMENTED** [review]
@ShubhamPalriwala thanks for the PR 😊 Just a small concern:

File: apps/web/app/(app)/environments/[environmentId]/settings/billing/actions.ts (hunk 1 above)
Comment (ShubhamPalriwala) [reply apps/web/app/(app)/environments/[environmentId]/settings/billing/actions.ts]: ahh apologies! it already happens inside the `updateTeam` service called inside the ee/billing method! Same for below, removed the both. Pushed 🤞🏼

**Review by mattinannt — APPROVED** [review]
💪 ready to merge 😊

**Pull request merged**